etna/insights/models.py

```diff
@@ -105,6 +106,31 @@ class InsightsPage(HeroImageMixin, TeaserImageMixin, BasePage):
         index.SearchField("insight_tag_names"),
     ]
 
+    def get_datalayer_data(self, request: HttpRequest) -> Dict[str, Any]:
+        data = super().get_datalayer_data(request)
+        if customDimension4 := self.topic:
+            pass
+        else:
+            customDimension4 = ""
```

Comment:
When the variable is something you want to access outside of the immediate conditional, there are clearer ways to write this sort of thing. For example, if there were more than two potential options, you could write like this:

```
some_variable = ""
if some_conditional:
    some_variable = some_value
elif some_other_conditional:
    some_variable = some_other_value
``` 

However, when the conditional logic is as simple as this, it's more readable to write things in a single line using an in-line conditional. For example:

```suggestion
        customDimension4 = self.topic.title if self.topic else ""
```

Also note: I'm also explicitly using `self.topic.title` as the value in the above example, as that is the string we want to display.

Reply:
Since `data` already has a default value for all dimensions, you could simplify this a bit, so that you don't need the `data.update()` at the end... For example:
```suggestion
        if self.topic:
            data["customDimension4"] = self.topic.title
```